From 202f41a8344d85673d936efbf8f38731b1ad706e Mon Sep 17 00:00:00 2001
From: Chris Blake <chrisrblake93@gmail.com>
Date: Tue, 23 Jul 2019 13:45:09 -0500
Subject: [PATCH 13/21] arm64: dts: sun50i-h5: Adjust OPP for nano boards with
 GPIO regulator

Make sure we set our voltages and settings for these correctly.
---
 .../allwinner/sun50i-h5-nanopi-neo-plus2.dts  | 53 +++++++++++++++-
 .../dts/allwinner/sun50i-h5-nanopi-neo2.dts   | 61 +++++++++++++++++++
 2 files changed, 113 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts
index 4f9ba53ffaae..39e648165fe5 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts
@@ -63,7 +63,7 @@
 		regulator-min-microvolt = <1100000>;
 		regulator-max-microvolt = <1300000>;
 		regulator-ramp-delay = <50>; /* 4ms */
-		gpios = <&r_pio 0 6 GPIO_ACTIVE_HIGH>;
+		gpios = <&r_pio 0 6 GPIO_ACTIVE_HIGH>; /* GPIOL6 */
 		gpios-states = <0x1>;
 		states = <1100000 0>, <1300000 1>;
 	};
@@ -83,6 +83,57 @@
 	status = "okay";
 };
 
+&cpu0 {
+  cpu0-supply = <&vdd_cpux>;
+  cooling-max-level = <12>;
+};
+
+/* We need to re-define this, as we do not have a proper voltage regulator */
+&cpu0_opp_table {
+  opp-120000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-240000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-480000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-648000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-816000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-960000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-1008000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-1056000000 {
+    opp-microvolt = <1300000 1300000 1300000>;
+  };
+  opp-1104000000 {
+    opp-microvolt = <1300000 1300000 1300000>;
+  };
+  opp-1152000000 {
+    opp-microvolt = <1300000 1300000 1300000>;
+  };
+  opp-1200000000 {
+    opp-microvolt = <1300000 1300000 1300000>;
+  };
+  opp-1224000000 {
+    opp-microvolt = <1300000 1300000 1300000>;
+  };
+  opp-1248000000 {
+    opp-microvolt = <1300000 1300000 1300000>;
+  };
+  /delete-node/ opp-1296000000;
+  /delete-node/ opp-1344000000;
+  /delete-node/ opp-1368000000;
+};
+
 &ehci0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo2.dts b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo2.dts
index b059e20813bd..e56bfb849771 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo2.dts
@@ -60,6 +60,67 @@
 		gpio = <&r_pio 0 2 GPIO_ACTIVE_HIGH>; /* PL2 */
 		status = "okay";
 	};
+
+  vdd_cpux: gpio-regulator {
+		compatible = "regulator-gpio";
+		regulator-name = "vdd-cpux";
+		regulator-type = "voltage";
+		regulator-boot-on;
+		regulator-always-on;
+		regulator-min-microvolt = <1100000>;
+		regulator-max-microvolt = <1300000>;
+		regulator-ramp-delay = <50>; /* 4ms */
+		gpios = <&r_pio 0 6 GPIO_ACTIVE_HIGH>; /* GPIOL6 */
+		gpios-states = <0x1>;
+		states = <1100000 0>, <1300000 1>;
+	};
+};
+
+&codec {
+	allwinner,audio-routing =
+		"Line Out", "LINEOUT",
+		"MIC1", "Mic",
+		"Mic",  "MBIAS";
+	status = "okay";
+};
+
+&cpu0 {
+  cpu0-supply = <&vdd_cpux>;
+  cooling-max-level = <6>;
+};
+
+/* We need to re-define this, as we do not have a proper voltage regulator */
+&cpu0_opp_table {
+  opp-120000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-240000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-480000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-648000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-816000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-960000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  opp-1008000000 {
+    opp-microvolt = <1100000 1100000 1100000>;
+  };
+  /delete-node/ opp-1056000000;
+  /delete-node/ opp-1104000000;
+  /delete-node/ opp-1152000000;
+  /delete-node/ opp-1200000000;
+  /delete-node/ opp-1224000000;
+  /delete-node/ opp-1248000000;
+  /delete-node/ opp-1296000000;
+  /delete-node/ opp-1344000000;
+  /delete-node/ opp-1368000000;
 };
 
 &ehci0 {
-- 
2.17.1

